core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -1123,10 +1134,21 @@ public AbstractProject getBuildingDownstream() {
      * the given project (provided that all builds went smoothly.)
      */
     public AbstractProject getBuildingUpstream() {
-        Set<Task> unblockedTasks = Jenkins.get().getQueue().getUnblockedTasks();
```

Comment:
please ensure any javadoc is updated to match the expected contract.
the change here is that if the upstream is blocked this no longer is a potential candidate.  (the previous javadoc may also have been incorrect)

e.g.
```
    /**
     * Returns the project if any of the upstream project is either
     * building, or scheduled and not blocked by an upstream/downstream project build.
     * <p>
     * This means eventually there will be an automatic triggering of
     * the given project (provided that all builds went smoothly.)
     */
```

Reply:
Done